### PR TITLE
add logic for exec transfer without log

### PR DESCRIPTION
**Pull request by lzxm160**

### CODE REVIEW

File: iotex-client/iotex-client.go [hunk 1]
```diff
@@ -322,8 +319,20 @@ func (c *grpcIoTexClient) decodeAction(ctx context.Context, act *iotextypes.Acti
 		return
 	}
 
-	if act.GetCore().GetExecution() != nil {
-		// TODO test when testnet enable systemlog
+	if act.GetCore().GetExecution() != nil && act.GetCore().GetExecution().GetAmount() != "0" && status == StatusSuccess {
```
Comment (dustinxie) [inline iotex-client/iotex-client.go:322]: i think we don't need amount != "0" here
should be if act.GetCore().GetExecution() != nil && status == StatusSuccess

File: iotex-client/iotex-client.go [hunk 2]
```diff
@@ -322,8 +319,20 @@ func (c *grpcIoTexClient) decodeAction(ctx context.Context, act *iotextypes.Acti
...
+		if len(receipt.Logs) == 0 {
```
Comment (dustinxie) [inline iotex-client/iotex-client.go:323]: and this is to handle pure transfer, so:
if act.GetCore().GetExecution().GetAmount() != "0"

File: iotex-client/iotex-client.go [hunk 3]
```diff
@@ -350,19 +348,50 @@ func (c *grpcIoTexClient) decodeAction(ctx context.Context, act *iotextypes.Acti
...
+	err = c.packTransaction(ret, src, dstAll, actionType, status, 1)
 	return
 }
 
-func (c *grpcIoTexClient) handleExecution(ctx context.Context, ret *types.Transaction, status, hash string, client iotexapi.APIServiceClient) (err error) {
+func (c *grpcIoTexClient) handleExecution(ctx context.Context, ret *types.Transaction, act *iotextypes.Action, h hash.Hash256, client iotexapi.APIServiceClient, callerAddr address.Address, status string) (err error) {
+	amount := act.GetCore().GetExecution().GetAmount()
+	if amount != "0" {
+		amount = "-" + amount
+	}
```
Comment (dustinxie) [inline iotex-client/iotex-client.go:359]: else {
  return
}
here?
that is, if amount = 0, then cannot have any token transfer in this tx, is that right?

Comment (yutongp) [reply iotex-client/iotex-client.go:359]: no. 0 in amount can still trigger contract run some code which make contract transfer token out of itself.
0 in amount only means no token out of sender's balance

File: iotex-client/iotex-client.go [hunk 4]
```diff
@@ -350,19 +348,50 @@ func (c *grpcIoTexClient) decodeAction(ctx context.Context, act *iotextypes.Acti
...
+	// deal with pure transfer to contract address
+	src := []*addressAmount{{
+		address: callerAddr.String(),
+		amount:  amount,
+	}}
```
Comment (yutongp) [inline iotex-client/iotex-client.go:364]: maybe doesn't include this if amount is 0

File: iotex-client/iotex-client.go [hunk 5]
```diff
@@ -350,19 +348,50 @@ func (c *grpcIoTexClient) decodeAction(ctx context.Context, act *iotextypes.Acti
...
+		if err != nil {
+			return err
+		}
+		contractAddr = responseReceipt.GetReceiptInfo().GetReceipt().GetContractAddress()
+	}
+
+	dst := []*addressAmount{{
+		address: contractAddr,
+		amount:  act.GetCore().GetExecution().GetAmount(),
+	}}
```
Comment (yutongp) [inline iotex-client/iotex-client.go:379]: same here. maybe doesn't include this if amount is 0

Comment (lzxm160) [reply iotex-client/iotex-client.go:379]: let me try this

**Pull request merged**